src/lava/magma/core/model/py/ports.py

```diff
@@ -176,27 +398,44 @@ class PyRefPort(AbstractPyPort):
 
     def __init__(self,
                  csp_send_port: ty.Optional[CspSendPort],
-                 csp_recv_port: ty.Optional[CspRecvPort], *args):
+                 csp_recv_port: ty.Optional[CspRecvPort],
+                 process_model: AbstractProcessModel,
+                 shape: ty.Tuple[int, ...] = tuple(),
+                 d_type: type = int):
         self._csp_recv_port = csp_recv_port
         self._csp_send_port = csp_send_port
-        super().__init__(*args)
+        super().__init__(process_model, shape, d_type)
 
     @property
     def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
+        """Property to get the corresponding CSP Ports of all connected
+        PyPorts (csp_ports). The CSP Port is the low level interface of the
+        backend messaging infrastructure which is used to send and receive data.
+
+        Returns
+        -------
+        A list of all CSP Ports connected to the PyPort.
+        """
         if self._csp_send_port is not None and self._csp_recv_port is not None:
             return [self._csp_send_port, self._csp_recv_port]
         else:
-            # In this case the port was not connected
+            # In this case the Port was not connected
             return []
 
+    @abstractmethod
     def read(
             self,
     ) -> ty.Union[
         np.ndarray, ty.Tuple[np.ndarray, np.ndarray], int, ty.Tuple[int, int]
     ]:
+        """Abstract method to request and return data from a VarPort.
+        Returns
+        -------
+        The value of the referenced var.
+        """
```

Comment:
I would always add the data type of the return var.

Reply:
The datatype is specified by the type hint, so it should not be necessary.